app/models/function.rb

```diff
@@ -0,0 +1,30 @@
+class Function < ApplicationRecord
+  belongs_to :user
+  belongs_to :role
+
+  before_save :correct_time_order?
```

Comment:
@elgrillo83 can you check this? 
http://guides.rubyonrails.org/active_record_validations.html#custom-methods

When I modify the validation methods to add an error message into the record it should be fine. Are you with me?